src/General/Alert/Alert.test.tsx

```diff
@@ -0,0 +1,263 @@
+    const alert = queryByRole('alertdialog');
+    fireEvent.keyDown(alert, { key: 'Enter', keyCode: 13 });
+    expect(onClose).toHaveBeenCalled();
+  });
+
+  test('press esc key should call onClose', async () => {
+    const onClose = jest.fn();
+    const { queryByRole } = render(<AlertComponent onClose={onClose} />);
+    const alert = queryByRole('alertdialog');
+    fireEvent.keyDown(alert, { key: 'Esc', keyCode: 27 });
+    expect(onClose).toHaveBeenCalled();
+  });
+});
+
+describe('<Alert/> autoClose', () => {
+  test('onClose should be called after isOpen changed from truthy to falsy', async () => {
+    const AlertComponent = ({
+      onClose,
+      isOpen,
+    }: {
+      onClose: () => void;
+      isOpen: boolean;
+    }) => (
+      <Alert
+        type={'success'}
+        message="You have successfully applied to Glints as Software Engineer."
+        isOpen={isOpen}
+        onClose={onClose}
+        autoClose={10}
+      />
+    );
+
+    const onClose = jest.fn();
+    const { rerender } = render(
+      <AlertComponent onClose={onClose} isOpen={false} />
+    );
+    rerender(<AlertComponent onClose={onClose} isOpen={true} />);
+    expect(onClose).not.toHaveBeenCalled();
+    await wait(() => {
+      // after autoClose timeout onClose should have been called
+      expect(onClose).toHaveBeenCalled();
+    });
+  });
+});
+
+describe('<Alert/> timeout', () => {
+  const AlertComponent = ({ isOpenProp }: { isOpenProp: boolean }) => {
+    const [isOpen, setIsOpen] = React.useState(isOpenProp);
+    return (
+      <Alert
+        type={'success'}
+        message="You have successfully applied to Glints as Software Engineer."
+        isOpen={isOpen}
+        onClose={() => setIsOpen(false)}
+      />
+    );
+  };
+
+  test('click close button should close alert after timeout', async () => {
+    const { queryByRole } = render(<AlertComponent isOpenProp={true} />);
+    const alert = queryByRole('alertdialog');
+    const closeButton = queryByRole('button');
+    fireEvent.click(closeButton);
+    expect(alert).toBeInTheDocument();
+    await wait(() => {
+      expect(alert).not.toBeInTheDocument();
+    });
+  });
+
+  test('press enter key should close alert after timeout', async () => {
+    const { queryByRole } = render(<AlertComponent isOpenProp={true} />);
+    const alert = queryByRole('alertdialog');
+    fireEvent.keyDown(alert, { key: 'Enter', keyCode: 13 });
+    expect(alert).toBeInTheDocument();
+    await wait(() => {
+      expect(alert).not.toBeInTheDocument();
+    });
+  });
+
+  test('press esc key should close alert after timeout', async () => {
+    const { queryByRole } = render(<AlertComponent isOpenProp={true} />);
+    const alert = queryByRole('alertdialog');
+    fireEvent.keyDown(alert, { key: 'Esc', keyCode: 27 });
+    expect(alert).toBeInTheDocument();
+    await wait(() => {
+      expect(alert).not.toBeInTheDocument();
+    });
+  });
+
+  test('autoClose', async () => {
```

Comment:
Could you please explain this part?
Also didn't understand the tests about after time out? Why should pressing the close icon close it after the time out

Reply:
you can check how the component is implemented, it has a state `isVisible` to control animation and rendering. When `isOpen` changing from true to false, it sets `isVisible` to false in a 300ms timeout for running css transition.